### ISSUE TITLE
rails 5 support?

### ISSUE THREAD

**Issue by nurey** [dark-panda/activerecord-spatial#2]
Would be great to make this work in rails 5 :)

**Comment by nurey** [comment]
I have a changeset that gets me almost there but not quite, at https://github.com/zoocasa/activerecord-spatial/commit/9e33c5bac0ada51c0ebb124ef543bfe091243afc

The blocker is that SpatialReflection is getting wrapped inside a RuntimeReflection.

**Comment by nurey** [comment]
I have a changeset that gets us there –
https://github.com/zoocasa/activerecord-spatial/commit/bea9b56e8602b6312e261c18dd1016433dd09b0f

**Comment by dark-panda** [comment]
Cool, will try this out on the weekend and see if I can get it merged soon.

**Comment by nurey** [comment]
There are still a bunch of failing tests stemming from the fact that `where_values` is no longer in Rails 5. Any help with that would be appreciated.

**Comment by nurey** [comment]
I have an updated changeset with less failing tests at https://github.com/zoocasa/activerecord-spatial/commit/90f8ae6159bdc06e2eb19aaa570262269efaf98b

I still have 2 FAILures and 3 ERRORs

**Comment by dark-panda** [comment]
I've pushed up a refactored branch at https://github.com/dark-panda/activerecord-spatial/tree/rails-5 that drops support for versions of Rails prior to 5 and includes most fixes. There's still one outstanding issue involving an operation on a mixed geometry according to the tests, but that appears to be a PostGIS 2.2+ thing, so I'll have to see if that's a change in PostGIS that I need to handle or a bug in PostGIS acting odd.

**Comment by nurey** [comment]
Awesome! I'll check it out tomorrow.

**Comment by nurey** [comment]
It looks good, we are using it in production now.

**Comment by nurey** [comment]
There's an issue with rails 5.1.1. I'm looking into it now.

**Comment by nurey** [comment]
`last_chain_scope` method signature has changed after this commit 
 https://github.com/rails/rails/commit/47b2ddd34da7dfabcda06d7fca39c1ca03ea87cc

**Comment by nurey** [comment]
I'm not well versed with the rails internals in this context. @dark-panda if you know of a quick fix, let me know.

**Comment by dark-panda** [comment]
I'll just have to abstract the calls away and wrap them into the necessary places. Nothing serious.

**Comment by nurey** [comment]
hey @dark-panda any chance you can add rails 5.2 support? We are still on Rails 5.0 here.

**Comment by ilvez** [comment]
Currently fails on Rails 5.2 with `activerecord-spatial-2.0.0/lib/activerecord-spatial/associations/base.rb:26:in <class:Preloader>: uninitialized constant ActiveRecord::Associations::Preloader::HasMany (NameError)`.

It seems that this class was last in 5.1.7: https://apidock.com/rails/ActiveRecord/Associations/Preloader/HasMany

Any chance you would be adding support for 5.2+?

**Comment by nurey** [comment]
@dark-panda, Rails 6.0 has been released a couple days ago. How long do you think it will take to get this gem 6.0 ready? Depending on the effort, we'd be happy to throw in some financial incentive.

**Comment by dark-panda** [comment]
Good question. I’ll have to look and see. Been a while since I delved into the code. But I’ll take a look at what tests fail and go from there.

**Comment by dark-panda** [comment]
Please see the rails-6-0 branch for Rails 6.0 support. It Works On My Machine™ at least.

**Comment by nurey** [comment]
I'll check it out, thanks!

**Comment by nurey** [comment]
Getting the following error:
```
NoMethodError: super: no superclass method `preloader_for' for #<ActiveRecord::Associations::Preloader:0x00007fe567700ec8>
        Did you mean?  preloaders_on
```

**Comment by dark-panda** [comment]
Do you have a test case and a backtrace?

**Comment by nurey** [comment]
See below:

```
From 85073b2513a616032ca2ea3d313326de390021d5 Mon Sep 17 00:00:00 2001
From: Ilia Lobsanov <ilia@nurey.com>
Date: Mon, 26 Aug 2019 12:40:24 -0400
Subject: [PATCH] Add test for includes

---
 test/associations_tests.rb | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/test/associations_tests.rb b/test/associations_tests.rb
index 0a919f7..e7e650e 100644
--- a/test/associations_tests.rb
+++ b/test/associations_tests.rb
@@ -65,6 +65,20 @@ class RelationshipsTest < ActiveRecordSpatialTestCase
   end
 end
 
+class PlainRelationshipsTest < ActiveRecordSpatialTestCase
+  def setup
+    load_models(:foo, :blort)
+  end
+
+  def test_with_eager_loading
+    values = Foo.includes(:blorts).to_a.collect do |foo|
+      foo.blorts.length
+    end
+
+    assert_equal([3, 3, 3], values)
+  end
+end
+
 class RelationshipsWithSelfTest < ActiveRecordSpatialTestCase
   def setup
     load_models(:foo, :bar)
-- 
2.23.0


```

**Comment by dark-panda** [comment]
Should be fixed now.

**Comment by dark-panda** [comment]
So then... all good? Releaseable?

**Comment by nurey** [comment]
Hey, yes I think it's good to go for a release.

**Comment by dark-panda** [comment]
Fixed and released in version 3.0.0.
